### PR TITLE
fix: use full URL for workspace OG image

### LINKED ISSUE #2994: Bug: OG image for workspace not working on X
### Describe the bug

When you link to a workspace on X, the OG image doesn't generate.

image

### Steps to reproduce

You can see it here: 
https://x.com/BekahHW/status/1770222173669433830?s=20

### Browsers

Chrome, Safari

### Additional context (Is this in dev or production?)

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by brandonroberts**
## Description

- Generates a full URL for the workspace OG image instead of just the path

## Related Tickets & Documents

Fixes #2994 

## Steps to QA

1. Find a public workspace URL
2. Paste the deploy preview into X to compose a post
3. Note that OG image is displayed in the preview 

## Tier (staff will fill in)

- [ ] Tier 1
- [ ] Tier 2
- [ ] Tier 3
- [x] Tier 4

### CODE REVIEW

**Review by nickytonline — COMMENTED** [review]
Took it for a spin and it looks like a full URL doesn't fix it either. 😞 
CleanShot 2024-04-02 at 16 49 55

**Comment by nickytonline** [comment]
It works! Full URL ftw!

CleanShot 2024-04-02 at 17 06 33

**Comment by brandonroberts** [comment]
Works for me too! I used a similar pattern we've been using in other places.

image

File: pages/workspaces/[workspaceId]/index.tsx [hunk 1]
```diff
@@ -50,15 +52,22 @@ export const getServerSideProps = async (context: GetServerSidePropsContext) =>
 
   setCookie({ response: context.res, name: WORKSPACE_ID_COOKIE_NAME, value: workspaceId });
 
-  return { props: { workspace: data } };
+  const ogImage = new URL(
+    getWorkspaceOgImage(data as Workspace, range),
+    process.env.NEXT_PUBLIC_BASE_URL || "http://localhost:3000"
```
Comment (nickytonline) [inline pages/workspaces/[workspaceId]/index.tsx:57]: I know this is for dev, but I'd be explicit with the check for production URL. 😅

**Review by nickytonline — APPROVED** [review]
:shipit:

File: pages/workspaces/[workspaceId]/index.tsx (hunk 1 above)
Comment (brandonroberts) [reply pages/workspaces/[workspaceId]/index.tsx:57]: 👍  What's the best way to check?

Comment (brandonroberts) [reply pages/workspaces/[workspaceId]/index.tsx:57]: I have this also

```ts
  const host = context.req.headers.host?.includes("localhost")
    ? `http://${context.req.headers.host}`
    : process.env.NEXT_PUBLIC_BASE_URL;
  const ogImage = new URL(getWorkspaceOgImage(data as Workspace, range), host);
```

Comment (nickytonline) [reply pages/workspaces/[workspaceId]/index.tsx:57]: Actually the env var is undefined locally, so I think we're good.

**Review by bdougie — APPROVED** [review]
FWIW I built https://socialcarding.com for this use case. 
Screen Shot 2024-04-02 at 3 10 44 PM

**Pull request merged**